File: Typescript/String_example.ts
const main = () => {
  const text = '  Hello Python World  ';
  const csvText = 'apple,banana,grape';
  const mixed = 'pyTHon123';

  console.log('Original:', JSON.stringify(text));
  console.log('trim():', text.trim());
  console.log('toLowerCase():', text.toLowerCase());
  console.log('toUpperCase():', text.toUpperCase());
  console.log(
    'title():',
    text
      .toLowerCase()
      .replace(/\b\w/g, (char) => char.toUpperCase()),
  );
  console.log(
    'replace("World", "Developers"):',
    text.replace('World', 'Developers'),
  );
  console.log('indexOf("Python"):', text.indexOf('Python'));
  console.log('startsWith("  Hello"):', text.startsWith('  Hello'));
  console.log('endsWith("  "):', text.endsWith('  '));
  console.log(
    'count("o"):',
    [...text].filter((char) => char === 'o').length,
  );
  console.log();

  const words = csvText.split(',');
  console.log('split(","):', words);
  console.log('join with " | " :', words.join(' | '));
  console.log();

  console.log('isalpha() on "Hello":', /^[A-Za-z]+$/.test('Hello'));
  console.log('isdigit() on "12345":', /^\d+$/.test('12345'));
  console.log('isalnum() on mixed:', /^[A-Za-z0-9]+$/.test(mixed));
  console.log();

  const name = 'Daniel';
  const age = 25;
  console.log(`template string example: Name = ${name}, Age = ${age}`);
};

main();

File: Python/String_example.py
def main() -> None:
    text = "  Hello Python World  "
    csv_text = "apple,banana,grape"
    mixed = "pyTHon123"

    print("Original:", repr(text))
    print("strip():", text.strip())
    print("lower():", text.lower())
    print("upper():", text.upper())
    print("title():", text.title())
    print('replace("World", "Developers"):', text.replace("World", "Developers"))
    print('find("Python"):', text.find("Python"))
    print('startswith("  Hello"):', text.startswith("  Hello"))
    print('endswith("  "):', text.endswith("  "))
    print('count("o"):', text.count("o"))
    print()

    words = csv_text.split(",")
    print('split(","):', words)
    print('join with " | " :', " | ".join(words))
    print()

    print('isalpha() on "Hello":', "Hello".isalpha())
    print('isdigit() on "12345":', "12345".isdigit())
    print("isalnum() on mixed:", mixed.isalnum())
    print()

    name = "Daniel"
    age = 25
    print(f"f-string example: Name = {name}, Age = {age}")


if __name__ == "__main__":
    main()
